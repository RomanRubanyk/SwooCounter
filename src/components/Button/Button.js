import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

const ButtonContainer = styled.Pressable`
  width: 160px;
  justify-content: center;
  align-items: center;
  padding: 16px;
  background-color: #ffffff;
  border-radius: 24px
  `;

const StyledText = styled.Text`
  font-weight: 700;
  color: #000000;
  font-size: 24px;
  letter-spacing: 0;
`;

const Button = ({ title, onPress }) => {
  return (
    <ButtonContainer onPress={onPress}>
      <StyledText>
        {title}
      </StyledText>
    </ButtonContainer>
  );
};

Button.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

export default Button;
