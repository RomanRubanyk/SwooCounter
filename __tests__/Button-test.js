import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../src/components/Button/Button'

describe('<Button/>', () => {
  let component;

  test('should render correctly', () => {
    component = renderer.create(<Button title="Button test" onPress={() => {}} />);

    expect(component).toMatchSnapshot();
  });

  test('should render correctly with props', () => {
    component = renderer.create(
      <Button title={'Button test'} onPress={() => {}} />,
    );

    expect(component).toMatchSnapshot();
  });
});
