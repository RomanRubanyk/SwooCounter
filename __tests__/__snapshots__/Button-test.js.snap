// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 24,
        "justifyContent": "center",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "width": 160,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 24,
          "fontWeight": "700",
          "letterSpacing": 0,
        },
      ]
    }
  >
    Button test
  </Text>
</View>
`;

exports[`<Button/> should render correctly with props 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 24,
        "justifyContent": "center",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "width": 160,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 24,
          "fontWeight": "700",
          "letterSpacing": 0,
        },
      ]
    }
  >
    Button test
  </Text>
</View>
`;
